test/fixtures/detach.js

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env node
+import process from 'node:process';
+import {execa} from '../../index.js';
+
+const subprocess = execa('node', ['./test/fixtures/forever.js'], {detached: true});
+console.log(subprocess.pid);
+process.exit(0);
```

Comment:
Not sure if it was worth adding a comment in the code but there might be a bug in Node.js (https://github.com/nodejs/node/issues/40808) preventing us from using `process.exit()` (without 0) in ESM.

Reply:
I can reproduce the bug you are mentioning.
I don't think this deserves a code comment since using an explicit exit code instead of making it implicit seems a rather normal best practice.